sdk/openai/openai/test/utils/setup.ts

```diff
@@ -20,8 +26,86 @@ function assertEnvironmentVariable(key: string): string {
   return value;
 }
 
-export default function ({ provide }: TestProject): void {
-  for (const key of Object.values(EnvironmentVariableNames)) {
-    provide(key, assertEnvironmentVariable(key));
+function getAccountNamesAndEndpoints(): { accountName: string; endpoint: string }[] {
+  return [
+    {
+      accountName: assertEnvironmentVariable(EnvironmentVariableNames.ACCOUNT_NAME_COMPLETIONS),
+      endpoint: assertEnvironmentVariable(EnvironmentVariableNames.ENDPOINT_COMPLETIONS),
+    },
+    {
+      accountName: assertEnvironmentVariable(EnvironmentVariableNames.ACCOUNT_NAME_VISION),
+      endpoint: assertEnvironmentVariable(EnvironmentVariableNames.ENDPOINT_VISION),
+    },
+    {
+      accountName: assertEnvironmentVariable(EnvironmentVariableNames.ACCOUNT_NAME_AUDIO),
+      endpoint: assertEnvironmentVariable(EnvironmentVariableNames.ENDPOINT_AUDIO),
+    },
+  ];
+}
+
+async function listDeployments(
+  subId: string,
+  rgName: string,
+  accountName: string,
+): Promise<DeploymentInfo[]> {
+  const deployments: DeploymentInfo[] = [];
+  const mgmtClient = new CognitiveServicesManagementClient(createTestCredential(), subId);
+  for await (const deployment of mgmtClient.deployments.list(rgName, accountName)) {
+    const deploymentName = deployment.name;
+    const modelName = deployment.properties?.model?.name;
+    const modelVersion = deployment.properties?.model?.version;
+    const capabilities = deployment.properties?.capabilities;
+    const sku = deployment.sku;
+    const isActive = deployment.properties?.provisioningState === "Succeeded";
+    if (sku && isActive && deploymentName && modelName && modelVersion) {
+      deployments.push({
+        deploymentName,
+        model: { name: modelName, version: modelVersion },
+        sku,
+        capabilities: capabilities as DeploymentInfo["capabilities"],
+      });
+    }
+  }
+  return deployments;
+}
+
+export async function getDeployments(): Promise<ResourcesInfo> {
```

Comment:
I'm curious, would this set up still work in the browser for file access?

Reply:
This `setup` script runs only in nodejs before the actual tests start running.